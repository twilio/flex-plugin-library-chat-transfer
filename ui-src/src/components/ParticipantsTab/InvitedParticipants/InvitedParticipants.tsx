import { Stack, Card, Heading } from '@twilio-paste/core';
import { Template, templates } from '@twilio/flex-ui';

import { InvitedParticipant } from './InvitedParticipant/InvitedParticipant';
import { InvitedParticipantDetails } from '../../../flex-hooks/types/InvitedParticipantDetails';
import { StringTemplates } from '../../../flex-hooks/strings/ChatTransferStrings';
import React from 'react';
interface InvitedParticipantsProps {
  invitedParticipantDetails: InvitedParticipantDetails[];
  handleCancelInvite: (invitedParticipantDetails: InvitedParticipantDetails) => void;
}

export const InvitedParticipants = ({ invitedParticipantDetails, handleCancelInvite }: InvitedParticipantsProps) => {
  const invitedParticipants = invitedParticipantDetails.map((invitedParticipantDetail) => {
    const participantName = invitedParticipantDetail.targetName;
    const { inviteTargetType } = invitedParticipantDetail;
    return (
      <InvitedParticipant
        participantName={participantName}
        key={invitedParticipantDetail.invitesTaskSid}
        inviteTargetType={inviteTargetType}
        handleCancelInvite={() => handleCancelInvite(invitedParticipantDetail)}
      />
    );
  });

  return (
    <Card padding="space60">
      <Heading as="h2" variant="heading20">
        {/* <Template source={templates['ConvTransferInvitedParticipants']} /> */}
      </Heading>
      <Stack orientation="vertical" spacing="space20">
        {invitedParticipants}
      </Stack>
    </Card>
  );
};
