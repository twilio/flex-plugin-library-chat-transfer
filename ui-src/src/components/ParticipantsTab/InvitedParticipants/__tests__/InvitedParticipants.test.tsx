import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { InvitedParticipants } from '../InvitedParticipants';

describe('InvitedParticipants', () => {
  const mockInvitedParticipantDetails = [
    // ... mock details
  ];

  it('renders invited participants correctly', () => {
    const mockHandleCancelInvite = jest.fn();

    const { getByText } = render(
      <InvitedParticipants
        invitedParticipantDetails={mockInvitedParticipantDetails}
        handleCancelInvite={mockHandleCancelInvite}
      />
    );

    const invitedParticipantsHeader = getByText('Invited Participants:');
    expect(invitedParticipantsHeader).toBeInTheDocument();

    // Loop through each invited participant and check if their names are rendered
    mockInvitedParticipantDetails.forEach((invitedParticipantDetail) => {
      const participantNameElement = getByText(invitedParticipantDetail.targetName);
      expect(participantNameElement).toBeInTheDocument();
    });

    // Click the cancel button for each participant and check if the callback is called
    mockInvitedParticipantDetails.forEach((invitedParticipantDetail) => {
      const cancelButton = getByTitle(`Cancel Invite for ${invitedParticipantDetail.targetName}`);
      fireEvent.click(cancelButton);
      expect(mockHandleCancelInvite).toHaveBeenCalledWith(invitedParticipantDetail);
    });
  });
});
