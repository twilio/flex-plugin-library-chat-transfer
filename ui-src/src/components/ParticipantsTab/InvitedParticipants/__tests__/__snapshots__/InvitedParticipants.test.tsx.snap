// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitedParticipants renders invited participants correctly 1`] = `[Function]`;
